mateweather/mateweather-dialog.c

```diff
@@ -104,6 +104,7 @@ static void response_cb(MateWeatherDialog* dialog, gint id, gpointer data)
 
 	mateweather_dialog_update (dialog);
     } else {
+        g_object_unref(dialog->priv->provider);
```

Comment:
I would suggest to do this in dispose to make sure that it will really be freed in all cases...